appinventor/blocklyeditor/src/component_database.js

```diff
@@ -683,6 +732,15 @@ Blockly.ComponentDatabase.prototype.forEachOptionList = function(callback) {
   goog.object.forEach(this.optionLists_, callback);
 }
 
+/**
+ * Returns the asset filter associated with the given key, or undefined.
+ * @param {number} key The key associated with a given filter.
+ * @return {Array<string>=}
+ */
+Blockly.ComponentDatabase.prototype.getFilter = function(key) {
+  return this.filters_[key];
```

Comment:
If key is out of bounds we should probably return `[]` instead of causing an error to be thrown.

Reply:
If the key is out of bounds it will return undefined. That works in all situations where this is called from :D

Reply:
I think this was meant to go [here](https://github.com/BeksOmega/appinventor-sources/pull/11#discussion_r480084266):
> You should change the return type of Blockly.Blocks.Utilities.assetKeyToBlocklyType to match this function then as it is the only place that it is called from. This return allows for undefined but the caller does not.